### ISSUE TITLE
Driver does not compile on Kernel 5.4 (Debian 10)

### ISSUE THREAD

**Issue by killenb** [ChimeraTK/gpcieuni-basedriver#3]
Reported by @markus-fleck

gpcieuni is compiling and loading

cat pcieuni_drv.c.in | sed "{s/@PCIEUNI_PACKAGE_VERSION@/0.1.10/}" > pcieuni_drv.c
make -C /lib/modules/5.4.0-0.bpo.3-amd64/build V=1 M=/var/lib/dkms/pcieuni/0.1.10/build modules
make[1]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.
make[1]: Entering directory '/usr/src/linux-headers-5.4.0-0.bpo.3-amd64'
make -C /usr/src/linux-headers-5.4.0-0.bpo.3-amd64 -f /usr/src/linux-headers-5.4.0-0.bpo.3-common/Makefile modules
test -e include/generated/autoconf.h -a -e include/config/auto.conf || (                \
echo >&2;                                                       \
echo >&2 "  ERROR: Kernel configuration is invalid.";           \
echo >&2 "         include/generated/autoconf.h or include/config/auto.conf are missing.";\
echo >&2 "         Run 'make oldconfig && make prepare' on kernel src to fix it.";      \
echo >&2 ;                                                      \
/bin/false)

[... hier kamen ein paar compiler aufrufe ...]

  { echo  /var/lib/dkms/pcieuni/0.1.10/build/pcieuni_drv.o /var/lib/dkms/pcieuni/0.1.10/build/pcieuni_fnc.o /var/lib/dkms/pcieuni/0.1.10/build/pcieuni_ioctl_dma.o; echo; } > /var/lib/dkms/pcieuni/0.1.10/build/pcieuni.mod
{   echo /var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko; :; } \
| awk '!x[$0]++' - > /var/lib/dkms/pcieuni/0.1.10/build/modules.order
make -f /usr/src/linux-headers-5.4.0-0.bpo.3-common/scripts/Makefile.modpost
  sed 's/ko$/o/' /var/lib/dkms/pcieuni/0.1.10/build/modules.order | scripts/mod/modpost -m  -i ./Module.symvers -I /var/lib/dkms/pcieuni/0.1.10/build/Module.symvers -e /lib/modules/5.4.0-0.bpo.3-amd64/gpcieuni/Module.symvers -o /var/lib/dkms/pcieuni/0.1.10/build/Module.symvers     -s -T -
ERROR: "pcieuni_write_no_struct_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_bufferList_append" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_set_drvdata" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_bufferList_init" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_register_write32" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_release_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_get_brdnum" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_bufferList_get_free" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_buffer_create" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_ioctl_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_probe_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_setup_interrupt" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_open_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_bufferList_set_free" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_cleanup_module_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_remove_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_init_module_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_bufferList_clear" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
ERROR: "pcieuni_read_no_struct_exp" [/var/lib/dkms/pcieuni/0.1.10/build/pcieuni.ko] undefined!
make[3]: *** [/usr/src/linux-headers-5.4.0-0.bpo.3-common/scripts/Makefile.modpost:94: __modpost] Error 1
make[2]: *** [/usr/src/linux-headers-5.4.0-0.bpo.3-common/Makefile:1622: modules] Error 2
make[1]: *** [/usr/src/linux-headers-5.4.0-0.bpo.3-common/Makefile:179: sub-make] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.4.0-0.bpo.3-amd64'
make: *** [Makefile:14: all] Error 2

**Comment by phako** [comment]
We relied on a bug in DKMS here:

On Ubuntu 16, the clean-up command looks like:

```
rm -rf "$dkms_tree/$module/$module_version/build/*"
```

Which is trying to remove a file named "*", so the build directory survives the clean step

On Debian 10, this is fixed to:
```
rm -rf "$dkms_tree/$module/$module_version/build"
```

**Comment by phako** [comment]
First, rather hackish fix: 
```
diff --git a/Makefile b/Makefile
index 8ab24f8..d093040 100644
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ HEADER_INSTALL_DIR=/usr/local/include/gpcieuni
 all: configure-source-files
        make -C /lib/modules/$(KVERSION)/build V=1 M=$(PWD) modules
 #copy the symvers file so it survives the dkms cleanup
-       cp Module.symvers gpcieuni.symvers
+       cp Module.symvers /tmp/gpcieuni.symvers
 
 #Performs a dkms install
 install: dkms-prepare install-headers
diff --git a/dkms.post_install b/dkms.post_install
index 40214d5..a8dc8a3 100755
--- a/dkms.post_install
+++ b/dkms.post_install
@@ -23,4 +23,4 @@ BUILD_DIR=$DKMS_TREE/$MODULE/$MODULE_VERSION/build
 
 #install the symvers file
 test -d $SYMVERS_DIR || mkdir -p $SYMVERS_DIR
-cp $BUILD_DIR/${MODULE}.symvers $SYMVERS_DIR/Module.symvers
+cp /tmp/${MODULE}.symvers $SYMVERS_DIR/Module.symvers
```

**Comment by phako** [comment]
Fixed with 1b5911be370f058305aeb70e7c8225220414594f
